apis/python-app-api/app_api.py

```
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Kubos Corporation
+# Licensed under the Apache License, Version 2.0
+# See LICENSE file for details.
+
+"""
+Mission Application API for Python Mission Applications.
+"""
+
+import toml
+import socket
+import json
+
+SERVICE_CONFIG_PATH = "/home/system/etc/config.toml"
+UDP_BUFF_LEN = 1024
+DEFAULT_TIMEOUT = 10.0  # Seconds
+
+
+class Services:
+
+    def __init__(self, service_config_filepath=SERVICE_CONFIG_PATH):
+        self.config = toml.load(service_config_filepath)
+
+    def query(self, service, query, timeout=DEFAULT_TIMEOUT):
+
+        # Check inputs
+        if service not in self.config:
+            raise KeyError("Service not available.")
+        if type(query) not in [str, unicode]:
+            raise TypeError("Query must be str or unicode.")
+
+        # Lookup port/ip
+        ip = self.config[service]["addr"]["ip"]
+        port = self.config[service]["addr"]["port"]
+
+        # Talk to the server
+        response = self._udp_query(query, (ip, port), timeout)
+
+        # Format the response and detect errors
+        (data, errors) = self._format(response, service)
+
+        # Check for endpoint errors
+        if errors not in ([], None):
+            raise EnvironmentError(
```

@plauche I moved the error creation for endpoint errors higher up here so it would not be buried in a lower level function. 

As for whether or not it _should_ throw an error, I still think it absolutely should. I agree the errors returned by the GraphQL endpoints should absolutely be preserved and handled by the user. The only difference is _how_ the user can handle it. The user is able to ignore the error field if we simply return it, and will not know there are any errors unless they check it. If we raise and error and return the error in the exception description, we are still returning the information, but forcing the user to deal with the error. This is more following a philosophy making tools to keep the user from doing bad things. I think since we are expecting users that are novice coders to use this library, I want to make sure they are forced to use it correctly.